src/hooks/useModal.js

```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react';
+
+function useModal() {
+  const [modalState, setModalState] = useState({
+    visibility: false,
+    target: '',
+  });
+
+  const onHandleCancel = (state) => setModalState(state);
```

Comment:
```suggestion
  const changeModalState = (newModalState) => {
    if (typeof newModalState === 'object') {
      setModalState({ ...newModalState });
    }
  };
```

이렇게 함수를 만들고 setter로 제공하면 modalState에 예상치 못한 값(null, string...)이 들어올때 대응하기 수월합니다.(일종의 자유도를 제약함으로서 상태를 안전하게 관리하는 법입니다.)